rascaline/src/labels/keys.rs

```diff
@@ -32,7 +32,38 @@ impl KeysBuilder for CenterSpeciesKeys {
 }
 
 /// Compute a set of keys with two variables: the central atom species and a
-/// single neighbor atom species.
+/// all neighbor atom species within the whole system.
+pub struct CenterSingleNeighborsSpeciesKeysSystem {
+    /// Should we consider an atom to be it's own neighbor or not?
+    pub self_pairs: bool,
+}
+
+impl KeysBuilder for CenterSingleNeighborsSpeciesKeysSystem {
+    fn keys(&self, systems: &mut [Box<dyn System>]) -> Result<Labels, Error> {
+
+        let mut all_species_pairs = BTreeSet::new();
+        for system in systems {
+            for &species_first in system.species()? {
+                for &species_second in system.species()? {
+                    if species_first == species_second && !self.self_pairs {
+                        continue;
+                    }
```

Comment:
This logic is weird, species_first and species_second could be from two different atoms, so why is `self_pairs` controlling that?

Reply:
To be honest. I don't know why I put this their. Should work fine without this condition...

Reply:
This might be a copy paste from `CenterSingleNeighborsSpeciesKeys` (which has to deal with this case)

Reply:
Might be. It is gone now.